app/src/main/java/de/westnordost/streetcomplete/data/meta/CountryInfos.kt

```diff
@@ -0,0 +1,56 @@
+package de.westnordost.streetcomplete.data.meta
+
+import android.content.res.AssetManager
+import com.charleskorn.kaml.Yaml
+import de.westnordost.countryboundaries.CountryBoundaries
+import java.io.File
+import java.io.SequenceInputStream
+
+class CountryInfos(private val assetManager: AssetManager) {
+    private val countryInfoMap = HashMap<String, IncompleteCountryInfo?>()
+    private val defaultCountryInfo: IncompleteCountryInfo by lazy {
+        loadCountryInfo("default")
+    }
+
+    /** Get the info by a list of country codes sorted by size. I.e. DE-NI,DE,EU gets the info
+     * for Niedersachsen in Germany and uses defaults from Germany and from the European Union */
+    fun get(countryCodesIso3166: List<String>): CountryInfo =
+        CountryInfo(countryCodesIso3166.mapNotNull { get(it) } + defaultCountryInfo)
+
+    private fun get(countryCodeIso3166: String): IncompleteCountryInfo? {
+        if (!countryInfoMap.containsKey(countryCodeIso3166)) {
+            val info = load(countryCodeIso3166)
+            countryInfoMap[countryCodeIso3166] = info
+        }
+        return countryInfoMap[countryCodeIso3166]
+    }
+
+    private fun load(countryCodeIso3166: String): IncompleteCountryInfo? {
+        val countryInfosFiles = assetManager.list(BASEPATH)
+        if (countryInfosFiles?.contains("$countryCodeIso3166.yml") == true) {
+            return loadCountryInfo(countryCodeIso3166)
+        }
+        return null
+    }
+
+    private fun loadCountryInfo(countryCodeIso3166: String): IncompleteCountryInfo {
+        val filename = "$countryCodeIso3166.yml"
+        assetManager.open(BASEPATH + File.separator + filename).use { inputStream ->
+            val countryCode = countryCodeIso3166.split("-").first()
+            val stream = SequenceInputStream("countryCode: $countryCode\n".byteInputStream(), inputStream)
+
+            return Yaml.default.decodeFromStream(IncompleteCountryInfo.serializer(), stream)
+        }
+    }
+
+    companion object {
+        private const val BASEPATH = "country_metadata"
+    }
+}
+
+fun CountryInfos.getByLocation(
+    countryBoundaries: CountryBoundaries,
+    longitude: Double,
+    latitude: Double,
+): CountryInfo =
+    get(countryBoundaries.getIds(longitude, latitude) ?: emptyList())
```

Comment:
Shouldn't be in the same file, IMO, so that CountryInfos does not have an (implicit) dependency on CountryBoundaries. Mabye in `CountryInfosXt.kt` or in the /ktx/ directory? (Not sure if it would be cleaner to only have extensions of things outside this project in that directory or if any utility extensions should go there.)

Also, could just be named `get`

Reply:
Oh well nevermind. 

I think the best would really be if the extension functions were not all collected in one place but placed each where they belong functionally. So I guess this place is as good as any other.

Reply:
We already have `CountryInfo.kt` in the `ktx` directory. Can it go in there, or should I create a new `CountryInfos.kt` or `CountryInfosXt.kt` there?